opentech/apply/funds/views.py

```diff
@@ -812,3 +814,24 @@ def delete(self, request, *args, **kwargs):
         )
         response = super().delete(request, *args, **kwargs)
         return response
+
+
+@method_decorator(login_required, name='dispatch')
+class SubmissionPrivateMediaRedirectView(RedirectView):
+    def get_redirect_url(self, *args, **kwargs):
+        try:
+            file_name = args[0]
+            submission_id = int(file_name.split('/')[1])
+        except (IndexError, ValueError):
+            raise Http404
+
+        try:
+            ApplicationSubmission.objects.get(id=submission_id)
+        except ApplicationSubmission.DoesNotExist:
+            raise Http404
+
+        # If user can access submission detail view then show the media
+        can_access_detail_view = SubmissionDetailView.as_view()(self.request, pk=submission_id)
```

Comment:
this is doing a lot of work to process the entire request through the view. This would also allow post requests to be processed against the detail view through the endpoint which is a bit weird.

Reply:
Makes sense. The initial thought was not to copy all permission checks. PR updated. Thanks.